src/modal/modal-ref.ts

```diff
@@ -46,7 +46,7 @@ export class NgbModalRef {
    *
    * When a `TemplateRef` is used as the content or when the modal is closed, will return `undefined`.
    */
-  get componentInstance(): any {
+  get componentInstance(): T | undefined {
```

Comment:
> Hi @SDAdham,
> 
> Thank you for your PR! I am sorry for not answering earlier.
> 
> As stated by @ExFlo, it is already possible to style the content of the dialog without this PR by using the existing `modalDialogClass` property with a custom CSS class and some CSS rules applying to the `.modal-content` element inside such a class.
> 
> However, I understand that you want to be able to add any custom class to the element directly, without even having to write any CSS, for example, to add some utility class provided by bootstrap (such as `border-0` or `bg-info`), which, in my opinion is a valid use case.
> 
> At the same time, @ExFlo seems to be worried about adding too many properties to ng-bootstrap components when it is possible to avoid them. Note that in case you want to reuse bootstrap utility classes inside your own CSS rules, you can [use Sass](https://getbootstrap.com/docs/5.1/customize/sass/).
> 
> Looking at the code itself, I have added some remarks in case we choose to integrate this PR.
> 
> @ExFlo @fbasso @maxokorokov What do you think?

Even with sass, in angular, this gets reused many times, overriding or rewriting a class that already provided by bootstrap is not really a good to go neither a good practice, especially bootstrap updates kicks in and new changes always come in. Defining `scss` files when it's not necessary just to override a class is also not necessary nor valid way to go. a lot of our components do not have `scss` files attached, then why should we create many scss files for a feature just because we don't have to add 0.0000000001 processor effort to provide a class through the method?